Terminal.Gui/Core/KeyBindings.cs

```diff
@@ -0,0 +1,370 @@
+		public KeyBinding (Type view, Key inKey, Key outKey, string description = "", bool enabled = true)
+		{
+			if (view == null) {
+				throw new ArgumentNullException ("View cannot be null.", nameof (view));
+			}
+			View = view.Name;
+			InKey = inKey;
+			OutKey = outKey;
+			Description = description;
+			Enabled = enabled;
+		}
+
+		/// <summary>
+		/// Get the view name.
+		/// </summary>
+		public string View { get; }
+		/// <summary>
+		/// Get or sets the input key pressed.
+		/// </summary>
+		public Key InKey { get; set; }
+		/// <summary>
+		/// Gets or sets the desired output key.
+		/// </summary>
+		public Key OutKey { get; set; }
+		/// <summary>
+		/// Gets or sets the description for this key.
+		/// </summary>
+		public string Description { get; set; }
+		/// <summary>
+		/// Gets or sets the enable status.
+		/// </summary>
+		public bool Enabled { get; set; } = true;
+
+	}
+
+	/// <summary>
+	/// Manage all the <see cref="KeyBinding"/> collection used by the <see cref="Application.KeyBindings"/>
+	/// </summary>
+	public sealed class KeyBindings {
+		private class KeyBindingEqualityComparer : IEqualityComparer<KeyBinding> {
+			public bool Equals (KeyBinding x, KeyBinding y)
+			{
+				if (x.View == y.View && (x.InKey == y.InKey || x.OutKey == y.OutKey))
+					return true;
+				return false;
+			}
+
+			public bool ReferenceEquals (KeyBinding x, KeyBinding y)
+			{
+				if (x.View == y.View && x.InKey == y.InKey && x.OutKey == y.OutKey && x.Description == y.Description && x.Enabled == y.Enabled)
+					return true;
+				return false;
+			}
+
+			public int GetHashCode (KeyBinding obj)
+			{
+				if (obj == null)
+					throw new ArgumentNullException ();
+
+				int hCode = 0;
+				foreach (var c in obj.View) {
+					hCode += c;
+				}
+				return hCode.GetHashCode ();
+			}
+		}
+
+		/// <summary>
+		/// Initializes a default instance.
+		/// </summary>
+		public KeyBindings ()
+		{
+		}
+
+		/// <summary>
+		/// Initializes a instance with the necessary settings.
+		/// </summary>
+		/// <param name="view">The view.</param>
+		/// <param name="inKey">The input key pressed.</param>
+		/// <param name="outKey">The desired output key.</param>
+		/// <param name="description">The description.</param>
+		/// <param name="enabled"><c>true</c> if enabled, <c>false</c>otherwise.</param>
+		public KeyBindings (Type view, Key inKey, Key outKey, string description = "", bool enabled = true)
+		{
+			AddKey (view, inKey, outKey, description, enabled);
+		}
+
+		/// <summary>
+		/// Initializes a instance of a <see cref="KeyBinding"/> instance.
+		/// </summary>
+		/// <param name="keyBinding">The key binding.</param>
+		public KeyBindings (KeyBinding keyBinding)
+		{
+			AddKey (keyBinding);
+		}
+
+		/// <summary>
+		/// The views name with the enabled status.
+		/// </summary>
+		public List<(string View, bool Enabled)> Views { get; private set; } = new List<(string View, bool Enabled)> ();
```

Comment:
It might simplify the code if this was a `Dictionary<string,bool>`?

Reply:
I agree, thanks.